cmd/validate/validate.go

```diff
@@ -134,16 +134,18 @@ func validateFile(fileSystem afero.Fs, file string) filewalker.Result {
 		if err == io.EOF {
 			break
 		}
-		if warcRecord.Type() == gowarc.Warcinfo {
-			warcInfoId = warcRecord.WarcHeader().GetId(gowarc.WarcRecordID)
-		}
+
 		result.IncrRecords()
 		result.IncrProcessed()
 		if err != nil {
 			result.AddError(fmt.Errorf("rec num: %d, offset: %d, cause: %w", result.Records(), currentOffset, err))
 			break
 		}
 
+		if warcRecord.Type() == gowarc.Warcinfo {
+			warcInfoId = warcRecord.WarcHeader().GetId(gowarc.WarcRecordID)
+		}
+
```

Comment:
Can you add a positive test case for getting the recordID?

Reply:
The test iterates over every record and in every iteration the recordID is accessed.

Reply:
See https://github.com/nlnwa/warchaeology/pull/136#discussion_r1584153510, this comment is outdated if we simply delete the entire thing.